pulp_container/app/registry_api.py

```diff
@@ -392,7 +392,16 @@ def get_dr_push(self, request, path, create=False):
 
     def create_dr(self, path, request):
         with transaction.atomic():
-            repository = serializers.ContainerPushRepositorySerializer.get_or_create({"name": path})
+            try:
+                repository = serializers.ContainerPushRepositorySerializer.get_or_create(
+                    {"name": path}
+                )
+            except models.ContainerPushRepository.DoesNotExist:
+                raise RepositoryInvalid(
+                    name=path,
+                    message='Cannot create "' + path + '" push repository, another repository with '
+                    "the same name already exists.",
+                )
             distribution = serializers.ContainerDistributionSerializer.get_or_create(
                 {"base_path": path, "name": path}, {"repository": get_url(repository)}
             )
```

Comment:
We also need to catch an exception here.
```
pulp file distribution create --name test/mygroup/ubuntu --base-path test/mygroup/ubuntu
podman push localhost:5001/test/mygroup/ubuntu --tls-verify=false
```
```
[pulp]           | During handling of the above exception, another exception occurred:
[pulp]           | 
[pulp]           | Traceback (most recent call last):
[pulp]           |   File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
[pulp]           |     response = get_response(request)
[pulp]           |   File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
[pulp]           |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
[pulp]           |   File "/usr/local/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 56, in wrapper_view
[pulp]           |     return view_func(*args, **kwargs)
[pulp]           |   File "/usr/local/lib/python3.9/site-packages/rest_framework/viewsets.py", line 124, in view
[pulp]           |     return self.dispatch(request, *args, **kwargs)
[pulp]           |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
[pulp]           |     response = self.handle_exception(exc)
[pulp]           |   File "/src/pulp_container/pulp_container/app/registry_api.py", line 279, in handle_exception
[pulp]           |     response = super().handle_exception(exc)
[pulp]           |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
[pulp]           |     self.raise_uncaught_exception(exc)
[pulp]           |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
[pulp]           |     raise exc
[pulp]           |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
[pulp]           |     response = handler(request, *args, **kwargs)
[pulp]           |   File "/src/pulp_container/pulp_container/app/registry_api.py", line 773, in create
[pulp]           |     _, repository = self.get_dr_push(request, path, create=True)
[pulp]           |   File "/src/pulp_container/pulp_container/app/registry_api.py", line 373, in get_dr_push
[pulp]           |     distribution, repository = self.create_dr(path, request)
[pulp]           |   File "/src/pulp_container/pulp_container/app/registry_api.py", line 405, in create_dr
[pulp]           |     distribution = serializers.ContainerDistributionSerializer.get_or_create(
[pulp]           |   File "/src/pulpcore/pulpcore/app/serializers/base.py", line 381, in get_or_create
[pulp]           |     result = cls.Meta.model.objects.get(**natural_key)
[pulp]           |   File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 87, in manager_method
[pulp]           |     return getattr(self.get_queryset(), name)(*args, **kwargs)
[pulp]           |   File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 637, in get
[pulp]           |     raise self.model.DoesNotExist(
[pulp]           | pulp_container.app.models.ContainerDistribution.DoesNotExist: ContainerDistribution matching query does not exist.
```

Maybe, we should start monitoring both `get_or_create` calls. `django.core.exceptions.ObjectDoesNotExist` would do us a favour.